src/app/_components/CopyToClipboard.tsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import { useState } from 'react'
+
+import { Link } from '@phosphor-icons/react/dist/ssr'
+import * as Sentry from '@sentry/nextjs'
+import { useCopyToClipboard } from 'usehooks-ts'
+
+import { NOTIFICATION_DIALOG_DURATION_MS } from '@/constants/notificationDialogDuration'
+
+import { Icon } from '@/components/Icon'
+import { NotificationDialog } from '@/components/NotificationDialog'
+
+type CopyToClipboardProps = {
+  text: string
+}
+
+export function CopyToClipboard({ text }: CopyToClipboardProps) {
+  const [_, copy] = useCopyToClipboard()
+  const [isCopied, setIsCopied] = useState(false)
+
+  async function handleCopy(text: string) {
+    try {
+      await copy(text)
+      setIsCopied(true)
+      setTimeout(() => setIsCopied(false), NOTIFICATION_DIALOG_DURATION_MS)
+    } catch (error) {
+      console.error('Failed to copy!', error)
+      Sentry.captureException(error)
+    }
+  }
+
+  return (
+    <>
+      <NotificationDialog
+        isOpen={isCopied}
+        setIsOpen={setIsCopied}
+        title="Link copied to clipboard!"
```

Comment:
The notification message should probably also be passed as a prop. What is the thing to copy is not a link?